src/backend/distributed/deparser/deparse_extension_stmts.c

```diff
@@ -0,0 +1,127 @@
+/*-------------------------------------------------------------------------
+ *
+ * deparse_extension_stmts.c
+ *	  All routines to deparse extension statements.
+ *	  This file contains deparse functions for extension statement deparsing
+ *    as well as related helper functions.
+ *
+ * Copyright (c), Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "distributed/deparser.h"
+#include "lib/stringinfo.h"
+#include "nodes/pg_list.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+
+/* Local functions forward declarations for helper functions */
+static void AppendCreateExtensionStmt(StringInfo str, CreateExtensionStmt *stmt);
+static void AppendDropExtensionStmt(StringInfo buf, DropStmt *stmt);
+static void AppendExtensionNameList(StringInfo buf, List *objects);
+
+
+/*
+ * DeparseCreateExtensionStmt builds and returns a string representing the
+ * CreateExtensionStmt to be sent to worker nodes.
+ */
+const char *
+DeparseCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt)
+{
+	StringInfoData sql = { 0 };
+	initStringInfo(&sql);
+
+	AppendCreateExtensionStmt(&sql, createExtensionStmt);
+
+	return sql.data;
+}
+
+
+/*
+ * AppendCreateExtensionStmt appends a string representing the CreateExtensionStmt to a buffer
+ */
+static void
+AppendCreateExtensionStmt(StringInfo str, CreateExtensionStmt *createExtensionStmt)
+{
+	/*
```

Comment:
we typically do not start functions with comments. Can't we move relevant pieces of the comment into the function?

Reply:
I will work on it